salt/client/__init__.py

```diff
@@ -1283,6 +1283,14 @@ def get_iter_returns(
                 if raw['data']['return'] == {}:
                     continue
 
+                # if the minion throws an exception containing the word "return"
+                # the master will try to handle the string as a dict in the next
+                # step. Check if we have a string, log the issue and continue.
+                if isinstance(raw['data']['return'], six.string_types):
+                    log.error("unexpected return from minion: %s", raw)
+                    raw['data']['return'] = {}
```

Comment:
i don't think we want to overwrite this, because we still want to print out the exceptions from the minion return so people can troubleshoot issues.